### PR TITLE
RDM-9182: Add exception handler for SQL errors

### PR DESCRIPTION
**Pull request by tomxelliott**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-9182

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by sullivangareth — APPROVED** [review]

**Comment by tomxelliott** [comment]
Unit test case added as per @sullivangareth comment 👍

**Review by sullivangareth — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ccd/endpoint/exception/RestExceptionHandler.java [hunk 1]
```diff
@@ -40,4 +42,13 @@ public RestExceptionHandler(AppInsights appInsights) {
         LOG.warn("exception with message {}, {}, {}", errorMsg, HttpStatus.INTERNAL_SERVER_ERROR, request, e);
         return handleExceptionInternal(e, errorMsg, new HttpHeaders(), HttpStatus.INTERNAL_SERVER_ERROR, request);
     }
+
+    @ExceptionHandler(SQLException.class)
+    protected ResponseEntity<Object> handleSQLException(final SQLException e, final WebRequest request) {
+        final String errorMsg = "SQL Exception thrown during API operation, " + HttpStatus.INTERNAL_SERVER_ERROR;
+        appInsights.trackException(e);
+
+        LOG.warn(errorMsg);
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/endpoint/exception/RestExceptionHandler.java:51]: I think this should be an error log.

File: src/main/java/uk/gov/hmcts/ccd/endpoint/exception/RestExceptionHandler.java [hunk 2]
```diff
@@ -40,4 +42,13 @@ public RestExceptionHandler(AppInsights appInsights) {
         LOG.warn("exception with message {}, {}, {}", errorMsg, HttpStatus.INTERNAL_SERVER_ERROR, request, e);
         return handleExceptionInternal(e, errorMsg, new HttpHeaders(), HttpStatus.INTERNAL_SERVER_ERROR, request);
     }
+
+    @ExceptionHandler(SQLException.class)
+    protected ResponseEntity<Object> handleSQLException(final SQLException e, final WebRequest request) {
+        final String errorMsg = "SQL Exception thrown during API operation, " + HttpStatus.INTERNAL_SERVER_ERROR;
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/endpoint/exception/RestExceptionHandler.java:48]: Appending HttpStatus.INTERNAL_SERVER_ERROR at end of error message didn't look useful. Error message can just tell what. The fact that it's returned as a http-500 doesn't need to be mentioned.

File: src/main/java/uk/gov/hmcts/ccd/endpoint/exception/RestExceptionHandler.java [hunk 3]
```diff
@@ -40,4 +42,13 @@ public RestExceptionHandler(AppInsights appInsights) {
...
+        appInsights.trackException(e);
+
+        LOG.warn(errorMsg);
+        return handleExceptionInternal(e, errorMsg, new HttpHeaders(), HttpStatus.INTERNAL_SERVER_ERROR, request);
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/endpoint/exception/RestExceptionHandler.java:52]: Error message should be returned inside a json object. All responses should be json.

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Pull request merged**